core/arch/x86/mangle.c

```diff
@@ -3076,6 +3085,7 @@ mangle_mov_seg(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
         PRE(ilist, instr,
             instr_create_save_to_tls(dcontext, reg, tls_slots[reg - REG_XAX]));
         /* restore reg */
+        /* XXX i#3307: necessary to instr_set_translation_mangling_epilogue? */
```

Comment:
These move instructions should be ok to re-execute.

Reply:
Comment removed.